scrapy_zyte_api/handler.py

```diff
@@ -32,6 +33,20 @@ def __init__(
         self._session = create_session()
         self._encoding = "utf-8"
 
+    @classmethod
+    def from_crawler(cls, crawler):
+        zyte_api_key = crawler.settings.get("ZYTE_API_KEY") or os.getenv("ZYTE_API_KEY")
+        if not zyte_api_key:
+            logger.warning(
+                "'ZYTE_API_KEY' must be set in the spider settings or env var "
+                "in order for ScrapyZyteAPIDownloadHandler to work."
+            )
+            raise NotConfigured
+
+        logger.info(f"Using Zyte API Key: {zyte_api_key[:7]}")
+        os.environ["ZYTE_API_KEY"] = zyte_api_key
```

Comment:
What do you think about passing api_key as a parameter to AsyncClient when these is a Scrapy setting present, instead of modifying os.environ?

Reply:
though if you're validating os.getenv anyways, there is no big difference

Reply:
Example where it may matter: you start 2 spiders in the same process using CrawlerRunner, and they have different API keys configured via custom_settings. In this case it won't work if you're modifying os.environ, but it'd work if you pass an API key. I know, that's really an edge case :)

Reply:
Good point! We'll handle that edge case in any case. Thanks for raising this! Updated in https://github.com/scrapy-plugins/scrapy-zyte-api/pull/5/commits/510ae72d965994df428e1edcf18f55fa61c2e9ff